src/m4a.c

```diff
@@ -0,0 +1,1425 @@
+{
+    if (mplayInfo->ident == ID_NUMBER)
+    {
+        ++mplayInfo->ident;
+        mplayInfo->fadeOC = speed;
+        mplayInfo->fadeOI = speed;
+        mplayInfo->fadeOV = (64 << FADE_VOL_SHIFT) | TEMPORARY_FADE;
+        mplayInfo->ident = ID_NUMBER;
+    }
+}
+
+void m4aMPlayFadeIn(struct MusicPlayerInfo *mplayInfo, u16 speed)
+{
+    if (mplayInfo->ident == ID_NUMBER)
+    {
+        ++mplayInfo->ident;
+        mplayInfo->fadeOC = speed;
+        mplayInfo->fadeOI = speed;
+        mplayInfo->fadeOV = (0 << FADE_VOL_SHIFT) | FADE_IN;
+        mplayInfo->status &= ~MUSICPLAYER_STATUS_PAUSE;
+        mplayInfo->ident = ID_NUMBER;
+    }
+}
+
+void m4aMPlayImmInit(struct MusicPlayerInfo *mplayInfo)
+{
+    s32 trackCount = mplayInfo->trackCount;
+    struct MusicPlayerTrack *track = mplayInfo->tracks;
+
+    while (trackCount > 0)
+    {
+        if (track->flags & MPT_FLG_EXIST)
+        {
+            if (track->flags & MPT_FLG_START)
+            {
+                Clear64byte(track);
+                track->flags = MPT_FLG_EXIST;
+                track->bendRange = 2;
+                track->volX = 64;
+                track->lfoSpeed = 22;
+                track->tone.type = 1;
+            }
+        }
+
+        --trackCount;
+        ++track;
+    }
+}
+
+void MPlayExtender(struct CgbChannel *cgbChans)
+{
+    struct SoundInfo *soundInfo;
+    u32 ident;
+
+    REG_SOUNDCNT_X = SOUND_MASTER_ENABLE
+                   | SOUND_4_ON
+                   | SOUND_3_ON
+                   | SOUND_2_ON
+                   | SOUND_1_ON;
+    REG_SOUNDCNT_L = 0; // set master volume to zero
+    REG_NR12 = 0x8;
+    REG_NR22 = 0x8;
+    REG_NR42 = 0x8;
+    REG_NR14 = 0x80;
+    REG_NR24 = 0x80;
+    REG_NR44 = 0x80;
+    REG_NR30 = 0;
+    REG_NR50 = 0x77;
+    soundInfo = SOUND_INFO_PTR;
+    ident = soundInfo->ident;
+    if (ident != ID_NUMBER)
+        return;
+    ++soundInfo->ident;
+    gMPlayJumpTable[8] = ply_memacc;
+    gMPlayJumpTable[17] = ply_lfos;
+    gMPlayJumpTable[19] = ply_mod;
+    gMPlayJumpTable[28] = ply_xcmd;
+    gMPlayJumpTable[29] = ply_endtie;
+    gMPlayJumpTable[30] = SampleFreqSet;
+    gMPlayJumpTable[31] = TrackStop;
+    gMPlayJumpTable[32] = FadeOutBody;
+    gMPlayJumpTable[33] = TrkVolPitSet;
+    soundInfo->cgbChans = cgbChans;
+    soundInfo->CgbSound = CgbSound;
+    soundInfo->CgbOscOff = CgbOscOff;
+    soundInfo->MidiKeyToCgbFreq = MidiKeyToCgbFreq;
+    soundInfo->maxLines = MAX_LINES;
+    CpuFill32(0, cgbChans, sizeof(struct CgbChannel) * 4);
+    cgbChans[0].ty = 1;
+    cgbChans[0].panMask = 0x11;
+    cgbChans[1].ty = 2;
+    cgbChans[1].panMask = 0x22;
+    cgbChans[2].ty = 3;
+    cgbChans[2].panMask = 0x44;
+    cgbChans[3].ty = 4;
+    cgbChans[3].panMask = 0x88;
+    soundInfo->ident = ident;
+}
+
+void MusicPlayerJumpTableCopy(void)
+{
+    asm("swi 0x2A");
+}
+
+void ClearChain_rev(void *x) //����������ΪClearChain,��Ϊm4a_asm.asm��Ҳ��һ��ClearChain(��katam�н�realClearChain)
```

Comment:
Even with the current script you can still rename both of them. 
e.g. 
`ClearChain_rev` -> `WhateverNameWithoutCollision`
`ClearChain` -> `RealClearChain` // `ClearChain_rev` won't be affected since it's already replaced
`WhateverNameWithoutCollision` -> `ClearChain`